samples/snippets/conftest.py

```diff
@@ -43,6 +45,11 @@ def project_id(bigquery_client: bigquery.Client) -> str:
     return bigquery_client.project
 
 
+@pytest.fixture(autouse=True)
+def reset_session():
```

Comment:
Might want a comment or docstring here explaining the purpose.
```suggestion
def reset_session():
    """An autouse fixture ensuring each sample runs in a fresh session.

    This allows us to have samples that query data in different locations.
    """
```